### ISSUE TITLE
MeshCtrl.js does not output requested data.

### ISSUE THREAD

**Issue by DaanSelen** [Ylianst/MeshCentral#6256]
**Describe the bug**
I try to use meshctrl.js to request data from the server through either the reverse proxy and directly to the server. Locally and remote. My python code which uses the same WebSocket connection, CAN requests the data from the MeshCentral server.

**To Reproduce**
Use MeshCtrl.js to request any information from the server.
`node node_modules/meshcentral/meshctrl.js --url wss://[url]/control.ashx --loginuser [username] --loginpass [password] ListDevices`
This returns just a newline.

**Expected behavior**
I expected to get the result in any form, text or json.

**Screenshots**
![image](https://github.com/user-attachments/assets/5782a4f4-dc50-4f4a-905b-283ec826f6b8)

**Server Software (please complete the following information):**
 - OS: Debian 12
 - Virtualization: ESXi
 - Network: WAN
 - Version: Debian 12
 - Node: v18.19.0

**Client Device (please complete the following information):**
 - Device: Laptop and Virtualized Server
 - OS: Debian Linux and Windows
 - Network: Local and WAN
 - Browser: Node in Terminal
 - MeshCentralRouter Version: N.A

**Additional context**
The MeshCentral has been running for a short time.

**Your config.json file**
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment__": " Ansible managed: /home/<REDACTED>/ansible/development/templates/meshcentral/config.json.j2 modified on 2024-07-12 16:09:24 by <REDACTED> on <REDACTED> ",
  "settings": {
    "cert": "<REDACTED>",
    "TlsOffload": false,
    "TrustedProxy": "<REDACTED>",
    "WANonly": true,
    "LANonly": false,
    "_sessionKey": "_IGNORED",
    "port": 8443,
    "aliasPort": 443,
    "_redirPort": 8080,
    "_redirAliasPort": 80,
    "postgres": {
      "host": "<REDACTED>",
      "user": "<REDACTED>",
      "port": "5432",
      "password": "<REDACTED>",
      "database": "<REDACTED>"
    }
  },
  "domains": {
    "": {
      "title": "<REDACTED>",
      "title2": "Remote Control",
      "minify": false,
      "allowedOrigin": true,
      "newAccounts": false,
      "userNameIsEmail": true,
      "certUrl": "<REDACTED>",
      "terminal": { "_linuxshell": "login" }
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "<REDACTED>",
    "names": "<REDACTED>",
    "skipChallengeVerification": true,
    "production": true
  }
}
```

**Comment by si458** [comment]
the url should be `wss://mydomain.com` you dont need anything else
also running `node meshctrl.js` will output information include include what the url should look like

**Comment by DaanSelen** [comment]
Suddenly works on another system...

**Comment by DaanSelen** [comment]
> the url should be `wss://mydomain.com` you dont need anything else like `control.ashx` at the end of it also running `node meshctrl.js` will output help information include include what the url should look like
> 
> ```
> simon@testmc:~/meshcentral/MeshCentral$ node meshctrl.js 
> MeshCtrl performs command line actions on a MeshCentral server.
> Information at: https://meshcentral.com
> No action specified, use MeshCtrl like this:
> 
>   meshctrl [action] [arguments]
> 
> Supported actions:
>   Help [action]               - Get help on an action.
>   ServerInfo                  - Show server information.
>   UserInfo                    - Show user information.
>   ListUsers                   - List user accounts.
>   ListUserSessions            - List online users.
>   ListUserGroups              - List user groups.
>   ListDevices                 - List devices.
>   ListDeviceGroups            - List device groups.
>   ListUsersOfDeviceGroup      - List the users in a device group.
>   ListEvents                  - List server events.
>   LoginTokens                 - List, create and remove login tokens.
>   DeviceInfo                  - Show information about a device.
>   EditDevice                  - Make changes to a device.
>   RemoveDevice                - Delete a device.
>   Config                      - Perform operation on config.json file.
>   AddUser                     - Create a new user account.
>   EditUser                    - Change a user account.
>   RemoveUser                  - Delete a user account.
>   AddUserGroup                - Create a new user group.
>   RemoveUserGroup             - Delete a user group.
>   AddToUserGroup              - Add a user, device or device group to a user group.
>   RemoveFromUserGroup         - Remove a user, device or device group from a user group.
>   RemoveAllUsersFromUserGroup - Remove all users from a user group.
>   AddDeviceGroup              - Create a new device group.
>   RemoveDeviceGroup           - Delete a device group.
>   EditDeviceGroup             - Change a device group values.
>   MoveToDeviceGroup           - Move a device to a different device group.
>   AddUserToDeviceGroup        - Add a user to a device group.
>   RemoveUserFromDeviceGroup   - Remove a user from a device group.
>   AddUserToDevice             - Add a user to a device.
>   RemoveUserFromDevice        - Remove a user from a device.
>   SendInviteEmail             - Send an agent install invitation email.
>   GenerateInviteLink          - Create an invitation link.
>   Broadcast                   - Display a message to all online users.
>   ShowEvents                  - Display real-time server events in JSON format.
>   RunCommand                  - Run a shell command on a remote device.
>   Shell                       - Access command shell of a remote device.
>   Upload                      - Upload a file to a remote device.
>   Download                    - Download a file from a remote device.
>   DeviceOpenUrl               - Open a URL on a remote device.
>   DeviceMessage               - Open a message box on a remote device.
>   DeviceToast                 - Display a toast notification on a remote device.
>   GroupMessage                - Open a message box on remote devices in a specific device group.
>   GroupToast                  - Display a toast notification on remote devices in a specific device group.
>   DevicePower                 - Perform wake/sleep/reset/off operations on remote devices.
>   DeviceSharing               - View, add and remove sharing links for a given device.
>   AgentDownload               - Download an agent of a specific type for a device group.
>   Report                      - Create and show a CSV report.
> 
> Supported login arguments:
>   --url [wss://server]        - Server url, wss://localhost:443 is default.
>                               - Use wss://localhost:443?key=xxx if login key is required.
>   --loginuser [username]      - Login username, admin is default.
>   --loginpass [password]      - Login password OR Leave blank to enter password at prompt
>   --token [number]            - 2nd factor authentication token.
>   --loginkey [hex]            - Server login key in hex.
>   --loginkeyfile [file]       - File containing server login key in hex.
>   --logindomain [domainid]    - Domain id, default is empty, only used with loginkey.
>   --proxy [http://proxy:123]  - Specify an HTTP proxy.
> ```

It was the URL looks like.

**Comment by DaanSelen** [comment]
To further add information, if you program your own script using the meshcentral websocket api. IT DOES need `control.ashx`. im pretty sure. That is what caused the confusion.

**Comment by si458** [comment]
@DaanSelen yes indeed if ur writing ur own code you need to add the /control.ashx it's just the meshctrl.js that adds it automatically so u don't need it.
